contrib/format-esri/src/main/java/org/apache/drill/exec/store/esri/ShpBatchReader.java

```diff
@@ -0,0 +1,318 @@
+          writeTimeColumn(rowWriter, field.getName(), dataVal);
+          break;
+        case LOGICAL:
+          int logicalVal = (Boolean) dbfRow[i] ? 1 : 0;
+          writeBooleanColumn(rowWriter, field.getName(), logicalVal);
+          break;
+        case NUMERIC:
+          double numericVal = ((Number) dbfRow[i]).doubleValue();
+          if (field.getDecimalCount() == 0) {
+            writeIntColumn(rowWriter, field.getName(), (int) numericVal);
+          } else {
+            writeDoubleColumn(rowWriter, field.getName(), numericVal);
+          }
+          break;
+        default:
+          break;
+      }
+    }
+  }
+
+  private void writeStringColumn(TupleWriter rowWriter, String name, String value) {
+    int index = rowWriter.tupleSchema().index(name);
+    if (index == -1) {
+      ColumnMetadata colSchema = MetadataUtils.newScalar(name, TypeProtos.MinorType.VARCHAR, TypeProtos.DataMode.OPTIONAL);
+      index = rowWriter.addColumn(colSchema);
+    }
+    ScalarWriter colWriter = rowWriter.scalar(index);
+    colWriter.setString(value);
+  }
+
+  private void writeDoubleColumn(TupleWriter rowWriter, String name, double value) {
+    int index = rowWriter.tupleSchema().index(name);
+    if (index == -1) {
+      ColumnMetadata colSchema = MetadataUtils.newScalar(name, TypeProtos.MinorType.FLOAT8, TypeProtos.DataMode.OPTIONAL);
+      index = rowWriter.addColumn(colSchema);
+    }
+    ScalarWriter colWriter = rowWriter.scalar(index);
+    colWriter.setDouble(value);
+  }
+
+  private void writeBooleanColumn(TupleWriter rowWriter, String name, int value) {
+    int index = rowWriter.tupleSchema().index(name);
+    if (index == -1) {
+      ColumnMetadata colSchema = MetadataUtils.newScalar(name, TypeProtos.MinorType.INT, TypeProtos.DataMode.OPTIONAL);
+      index = rowWriter.addColumn(colSchema);
+    }
+    boolean boolean_value = true;
+    if(value == 0){
+      boolean_value = false;
+    }
+
+    ScalarWriter colWriter = rowWriter.scalar(index);
+    colWriter.setBoolean(boolean_value);
+  }
+
+  private void writeIntColumn(TupleWriter rowWriter, String name, int value) {
+    int index = rowWriter.tupleSchema().index(name);
+    if (index == -1) {
+      ColumnMetadata colSchema = MetadataUtils.newScalar(name, TypeProtos.MinorType.INT, TypeProtos.DataMode.OPTIONAL);
+      index = rowWriter.addColumn(colSchema);
+    }
+    ScalarWriter colWriter = rowWriter.scalar(index);
+    colWriter.setInt(value);
+  }
+
+  private void writeTimeColumn(TupleWriter rowWriter, String name, long value) {
+    int index = rowWriter.tupleSchema().index(name);
+    Instant instant = new Instant(value);
+    if (index == -1) {
+      ColumnMetadata colSchema = MetadataUtils.newScalar(name, TypeProtos.MinorType.INT, TypeProtos.DataMode.OPTIONAL);
+      index = rowWriter.addColumn(colSchema);
+    }
+    ScalarWriter colWriter = rowWriter.scalar(index);
+    colWriter.setTimestamp(instant);
+  }
+
+  @Override
+  public void close() {
```

Comment:
Please create separate private method that closes streams as you did in excel plugin.

Reply:
Please address this comment...

Reply:
Hi @arina-ielchiieva I'm not sure what you're getting at here.  The `close()` method is pretty much cut/paste from the Excel reader.  Are you looking for a separate method for each stream or object?

https://github.com/apache/drill/blob/f2654eed3b6d444280b5c0c2a507edc355c676e6/contrib/format-excel/src/main/java/org/apache/drill/exec/store/excel/ExcelBatchReader.java#L428-L447

Reply:
I wanted to extract common logic into senate method:
```
  @Override
  public void close() {
    closeStream(fileReaderShp, "ESRI Shapefile");
    closeStream(fileReaderDbf, "DBF Shapefile");
    closeStream(fileReaderPrj, "PRJ Shapefile");
    try {
      dbfReader.close();
    } catch (Exception e) {
      logger.warn("Error when closing DBF Reader: {}", e.getMessage());
    }
    dbfReader = null;
  }


  /**
   * Closes and nullifies given input stream.
   * In case of {@link IOException}, logs it as warning.
   *
   * @param inputStream input stream
   */
  private void closeStream(InputStream inputStream, String name) {
    if (inputStream == null) {
      return;
    }
    try {
      inputStream.close();
    } catch (IOException e) {
      logger.warn("Error when closing {}: {}", name, e.getMessage());
    }
    inputStream = null;
  }
```